### DISCUSSION TITLE
:grey_question: UNNEST list of (json) objects

### DISCUSSION THREAD

**Discussion by adriens** [duckdb/duckdb#6763] [Q&A]
Hi, I'm currently trying to transform this : 

<pre>────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│                                                                                                                     j                                                                                                                      │
│                                                                                                                    json                                                                                                                    │
├────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ [{&quot;name&quot;:&quot;os&quot;,&quot;uri&quot;:&quot;https://deploy-preview-2080--endoflife-date.netlify.app/api/v1/categories/os/&quot;},{&quot;name&quot;:&quot;lang&quot;,&quot;uri&quot;:&quot;https://deploy-preview-2080--endoflife-date.netlify.app/api/v1/categories/lang/&quot;},{&quot;name&quot;:&quot;service&quot;,&quot;uri&quot;:&quot;ht…  │
└────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
</pre>

to a table of rows... then into a final table.

But when I try to unnest, it tells be that we cannot unnest objects.

Any idea on how to handle this pattern around json ?

# :tickets: Related issues

- https://github.com/duckdb/duckdb/issues/2525

**Comment by Mause** [comment]
Can you please give a code sample that reproduces the issue? Ie, that doesn't contain ellipsis

**Reply by adriens** [reply]
```sql
install json;
load json;

create or replace table load_categories (j json);

insert into load_categories
    select *
        from read_json_objects('https://deploy-preview-2080--endoflife-date.netlify.app/api/v1/categories/');
update load_categories set j = json_extract(j, '$.result');
select * from load_categories;

-- Something here to unnest
-- ???
-- Target final table
create table categories (
        name VARCHAR NOT NULL PRIMARY KEY,
        uri VARCHAR NOT NULL UNIQUE,
);

-- Finallly the insert
-- insert into categories select * from .....
```

**Reply by Mause** [reply]
You probably want to use `read_json_auto` instead, that way the data is decomposed into DuckDB native types

Have a look at the docs for more info: https://duckdb.org/docs/extensions/json

**Comment by adriens** [comment]
Got it !!!

```sql
SELECT 
        p.name,
        p.uri    
    FROM (SELECT unnest(result) p FROM read_json_auto('https://deploy-preview-2080--endoflife-date.netlify.app/api/v1/categories/'))
```

**Reply by adriens** [reply]
thanks @Mause for the fast support :+1:

**Comment by adriens** [comment]
Btw, great resource that helped a lot : [Loading JSON Files using DuckDB](https://levelup.gitconnected.com/loading-json-files-using-duckdb-a93240961710)
